src/server/command_registry.cc

```diff
@@ -38,8 +46,26 @@ bool CommandId::IsTransactional() const {
   return false;
 }
 
+CommandRegistry::CommandRegistry() {
+  string rename_command = GetFlag(FLAGS_rename_command);
+  vector<std::string_view> parts = absl::StrSplit(rename_command, ",", absl::SkipEmpty());
+  for (std::string_view p : parts) {
+    size_t sep = p.find('=');
+
+    string cmd_name = string(p.substr(0, sep));
+    transform(cmd_name.begin(), cmd_name.end(), cmd_name.begin(), ::toupper);
+    string cmd_new_name = string(p.substr(sep + 1));
+    transform(cmd_new_name.begin(), cmd_new_name.end(), cmd_new_name.begin(), ::toupper);
+    cmd_rename_map_.emplace(cmd_name, cmd_new_name);
```

Comment:
how is emplace dangerous?
I dont mind using [] operator,  using it will have the affect in your example of a rename to c, which I am not sure this better (or worse), I dont want to have a documentation on this flag that says that if you use` a=b,a=c` that a will be set to c. I rather just exit in this case, and in this case using emplace has more sense to me

Reply:
Generally, `emplace*()` family is dangerous and can cause unintended behavior, due to the fact that it utilizes explicit constructors implicitly. There are a couple of examples here: https://stackoverflow.com/questions/10890653/why-would-i-ever-use-push-back-instead-of-emplace-back

Generally, `emplace()` is designed to be a faster `push()`/`insert()` method (though less safe in the general case) that avoids creating a temporary object.

BTW to speed things up, we could `std::move()` both `cmd_name` and `cmd_new_name` into the map. Again, it doesn't matter, and also I imagine that all (most?) commands will be under the SSO size anyway.

> I rather just exit in this case
What do you mean? exit as in `exit(-1);`? That's indeed something to consider (but unless I missed something, it's not implemented atm?)

BTW I think it would be more standard to just take the latest value, that's how most Unix command line tools work with regards to repeatedly used flags, but I agree with you that it does not matter.